omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_4/FrontendJsonConfigController2_4.java

```diff
@@ -0,0 +1,155 @@
+            }
+            String requestBody = stringBuilder.toString();
+
+            new ObjectMapper().readTree(requestBody); // verify that is in a valid JSON format
+
+            InputStream inputStream = new ByteArrayInputStream(requestBody.getBytes(StandardCharsets.UTF_8));
+            OutputStream outStream = Files.newOutputStream(jsonConfigFile.toPath());
+            OpenmrsUtil.copyFile(inputStream, outStream);
+
+            if (jsonConfigFile.exists()) {
+                log.debug("file: '{" + jsonConfigFile.getAbsolutePath() + "}' written successfully");
+                response.setStatus(HttpServletResponse.SC_OK);
+            }
+        } catch (JsonProcessingException e) {
+            log.error("Invalid JSON format", e);
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    private boolean isValidAuthFormat(HttpServletResponse response, String basicAuth) {
+        if (basicAuth.startsWith("Basic")) {
+            try {
+                // remove the leading "Basic "
+                basicAuth = basicAuth.substring(6);
+                if (StringUtils.isBlank(basicAuth)) {
+                    return true;
+                }
+
+                String decoded = new String(Base64.getDecoder().decode(basicAuth), StandardCharsets.UTF_8);
+                if (StringUtils.isBlank(decoded) || !decoded.contains(":")) {
+                    return true;
+                }
+
+                String[] userAndPass = decoded.split(":");
+                Context.authenticate(new UsernamePasswordCredentials(userAndPass[0], userAndPass[1]));
+                log.debug("authenticated [{" + userAndPass[0] + "}]");
+            } catch (Exception ex) {
+                // This filter never stops execution. If the user failed to
+                // authenticate, that will be caught later.
+                log.debug("authentication exception ", ex);
+            }
+        }
+        return false;
+    }
+
+        private File getJsonConfigFile() {
+        File folder = getSpaStaticFilesDir();
+        if (!folder.isDirectory()) {
+            log.warn("SPA frontend repository is not a directory hence creating it at: " + folder.getAbsolutePath());
+            if (!folder.mkdirs()) {
+                throw new ModuleException("Failed to create the SPA frontend repository folder at: " + folder.getAbsolutePath());
+            }
+        }
+        return new File(folder.getAbsolutePath(), JSON_CONFIG_FILE_NAME);
+    }
+
+    private File getSpaStaticFilesDir() {
+        AdministrationService as = Context.getAdministrationService();
+        String folderName = as.getGlobalProperty(GP_LOCAL_DIRECTORY, DEFAULT_FRONTEND_DIRECTORY);
+
+        // try to load the repository folder straight away.
+        File folder = new File(folderName);
+
+        // if the property wasn't a full path already, assume it was intended to be a
+        // folder in the
+        // application directory
```

Comment:
Is there any reason why the above comment cannot be in two lines instead of three?